resources/tentative/complex-static-html/src/big-dom-generator/components/tree-area.jsx

```diff
@@ -0,0 +1,69 @@
+import { LCG } from 'random-seedable';
+import { useRef } from 'react';
+import { SEED, MAX_DEPTH, MAX_BREADTH, CHILD_PROB, TARGET_SIZE } from './../params';
+import ChevronRight from './../assets/Smock_ChevronRight_18_N.svg';
+
+const FolderWrapper = (props) => {
+  const {nodeCount, rando, maxDepth, maxBreadth, childProb, currentDepth} = props;
+  if (currentDepth >= maxDepth) {
+    return (null);
+  }
+  // Choose a random number of children.
+  const numChildren = rando.randRange(1, maxBreadth);
+  const children = [];
+  for (let i = 0; i < numChildren && nodeCount.current < TARGET_SIZE; i++) {
+    children.push(<TreeItem key={i}
+                            nodeCount={nodeCount}
+                            rando={rando}
+                            numChildren={numChildren}
+                            maxBreadth={maxBreadth}
+                            maxDepth={maxDepth}
+                            childProb={childProb}
+                            currentDepth={currentDepth+1}/>);
+  }
+  nodeCount.current = nodeCount.current + 1;
+  return (
+    <ul className="ui spectrum-TreeView spectrum-TreeView--sizeM">
```

Comment:
I wonder if we should use `<details open>` instead of `<ul>` here. Potentially then we could have something like a maxVisibleDepth option where, once crossed, the details become closed. This may help with @rniwa's request for more hidden elements and also prevent the nest level from causing a large hidden overflow-x in the sidebar.

Reply:
Do people use details element in real life? I've personally never seen a real website that makes a heavy use of details elements.

Reply:
As a datapoint, I was just looking at a tree I made with `<details>` like https://bgrins.github.io/xbl-analysis/tree/2017-10-01.html and the structure is

```
<details>
  <summary>Item 1</summary>
  <ul>
    <li>
      <details>
        <summary>Item 1.1</summary>
      </details>
     </li>
    <li>
      <details>
        <summary>Item 1.2</summary>
      </details>
     </li>
   </ul>
</details>
```

So actually not too different from what the structure here, except an extra wrapper element, not having the custom chevron element, and likely removing some of the spectrum classes.

Reply:
But is that really a common pattern? I'd rather not use details element unless we know it's frequently used in wild. Websites like nytimes.com, gmail, cnn.com, etc... don't use details element.

Reply:
FWIW the pattern I used here is supported by the [treeview component](https://opensource.adobe.com/spectrum-css/treeview.html#foldersfiles) of adobe spectrum-css, so I'd guess it is a common pattern. Also, the library itself uses the `.is-open` class to control the visibility of the children via css. So we can just remove the `.is-open` for elements after `maxVisibleDepth`.

Reply:
Sure, that sounds reasonable. IMO whatever max visible depth which prevents things from being scrolled out of view of the sidebar would be good.

Reply:
Reduced the spacing, font and set a limit of 15 to open objects. See the [updated version](https://lpardosixtosms.github.io/complex-dom/)